src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml

```diff
@@ -883,10 +885,10 @@ let%test_unit "foreign_field_mul gadget" =
                   (module Runner.Impl)
                   product
               in
-              assert (product = expected) ) ;
+              assert Stdlib.(product = expected) ) ;
 
           (* TODO: 2) Add result bound addition gate *)
-          assert (1 == List.length external_checks.bounds) ;
+          assert (1 = List.length external_checks.bounds) ;
```

Comment:
```suggestion
          assert (Mina_stdlib.List.Length.equal external_checks.bounds 1) ;
```
List.length runs in O(n) where n is the length of the list. The function above runs in time the length of the cst you pass.

That also means you should add `Mina_stdlib` as dune library dependency.